src/app/services/firebase/firebase.service.ts

```diff
@@ -33,6 +33,12 @@ export abstract class FirebaseService {
     });
   }
 
+  updateDocument<T extends { id: string }>(collection: CollectionType, data: T): Observable<T> {
+    return new Observable<T>(observer => {
```

Comment:
use 'from' operator like in delete

Reply:
Why don't you use upsert? It should be the same

Reply:
removed